### PR TITLE
feat(DDM): add ddm accounts data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ddm accounts data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add ddm accounts data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ddm/' TESTARGS='-run TestAccDatasourceDdmAccounts_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ddm/ -v -run TestAccDatasourceDdmAccounts_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDdmAccounts_basic
=== PAUSE TestAccDatasourceDdmAccounts_basic
=== CONT  TestAccDatasourceDdmAccounts_basic
--- PASS: TestAccDatasourceDdmAccounts_basic (452.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ddm       452.186s
```

### CODE REVIEW

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		getDdmAccountsProduct = "ddm"
+	)
+	getDdmAccountsClient, err := cfg.NewServiceClient(getDdmAccountsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDM Client: %s", err)
+	}
+
+	getDdmAccountsPath := getDdmAccountsClient.Endpoint + getDdmAccountsHttpUrl
+	getDdmAccountsPath = strings.ReplaceAll(getDdmAccountsPath, "{project_id}", getDdmAccountsClient.ProjectID)
+	getDdmAccountsPath = strings.ReplaceAll(getDdmAccountsPath, "{instance_id}", fmt.Sprintf("%v", d.Get("instance_id")))
```
Comment (deer-hang) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go]: This line over 120 characters

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the description of the DDM account.`,
+			},
+			"schemas": {
+				Type:        schema.TypeList,
+				Elem:        AccountsAccountSchemaSchema(),
+				Computed:    true,
+				Description: `Specifies the schemas that associated with the account.`,
```
Comment (deer-hang) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go]: Replace `Specifies` with `Indicates`, check the other parts.

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go]: done

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/data_source_huaweicloud_ddm_accounts.go]: done

**Pull request merged**